test/OriginalTokenBridge.test.js

```diff
@@ -307,6 +307,51 @@ describe("OriginalTokenBridge", () => {
         })
     })
 
+    describe("sets withdrawal fee", () => {
+        const withdrawalFeeBps = 100
+        it("reverts when called by non owner", async () => {
+            await expect(originalTokenBridge.connect(user).setWithdrawalFeeBps(withdrawalFeeBps)).to.be.revertedWith("Ownable: caller is not the owner")
+        })
+
+        it("sets withdrawal fee", async () => {
+            await originalTokenBridge.setWithdrawalFeeBps(withdrawalFeeBps)
+            expect(await originalTokenBridge.withdrawalFeeBps()).to.be.eq(withdrawalFeeBps)
+        })
+    })
+
+    describe("bridges and withdraw fee", () => {
+        const withdrawalFeeBps = 100
+        let fee
+        let totalAmount
+        beforeEach(async () => {
+            await originalTokenBridge.setWithdrawalFeeBps(withdrawalFeeBps)
+            fee = (await originalTokenBridge.estimateBridgeFee(false, adapterParams)).nativeFee
+            totalAmount = amount + fee
+            await originalToken.connect(user).approve(originalTokenBridge.target, amount)
+        })
+
+        it("bridges ERC20 token and withdraws fees", async () => {
+            await originalTokenBridge.registerToken(originalToken.target, sharedDecimals)
+            await originalTokenBridge.connect(user).bridge(originalToken.target, amount, user.address, callParams, adapterParams, { value: fee })
+            const LDtoSD = await originalTokenBridge.LDtoSDConversionRate(originalToken.target)
+
+            const withdrawalFee = (amount * BigInt(withdrawalFeeBps)) / BigInt(10000) / LDtoSD
+
+            await originalTokenBridge.connect(owner).withdrawFee(originalToken.target, owner.address, withdrawalFee)
+            expect(await originalToken.balanceOf(owner.address)).to.be.eq(withdrawalFee)
```

Comment:
Can you check the following:
when fee is 1%,  while bridging X user receives 99%, the fees are 1%.
I see you only checked the fee part

Reply:
@leonprou I have updated the test which checks the total value locked for a token which is essentially the actual amount of token sent in the LZ message to be minted on the dest chain